app/actions/index.js

```diff
@@ -33,6 +33,119 @@ export const createDat = () => dispatch => {
   addDat({ path })(dispatch)
 }
 
+export const showDownloadScreen = key => ({
+  type: 'SHOW_DOWNLOAD_SCREEN',
+  key: encode(key)
+})
+export const hideDownloadScreen = () => ({ type: 'HIDE_DOWNLOAD_SCREEN' })
+export const cancelDownloadDat = key => dispatch => {
+  key = encode(key)
+  const { dat } = dats[key]
+
+  for (const con of dat.network.connections) {
+    con.removeAllListeners()
+  }
+  dat.stats.removeAllListeners()
```

Comment:
While it seems necessary that we do this, I kinda thought that `dat.close()` should do that 🤔

Reply:
that's not common in the node ecosystem

Reply:
that's not common in the node ecosystem